supply/lib/supply/client.rb

```diff
@@ -22,25 +22,28 @@ class Client
 
     # instantiate a client given the supplied configuration
     def self.make_from_config
-      unless Supply.config[:json_key] || (Supply.config[:key] && Supply.config[:issuer])
+      unless Supply.config[:json_key] || Supply.config[:json_key_data] || (Supply.config[:key] && Supply.config[:issuer])
         UI.important("To not be asked about this value, you can specify it using 'json_key'")
         Supply.config[:json_key] = UI.input("The service account json file used to authenticate with Google: ")
       end
 
       return Client.new(path_to_key: Supply.config[:key],
                         issuer: Supply.config[:issuer],
-                        path_to_service_account_json: Supply.config[:json_key])
+                        path_to_service_account_json: Supply.config[:json_key], service_account_json: Supply.config[:json_key_data])
     end
 
     # Initializes the android_publisher and its auth_client using the specified information
+    # @param service_account_json: The raw service account Json data
     # @param path_to_service_account_json: The path to your service account Json file
     # @param path_to_key: The path to your p12 file (@deprecated)
     # @param issuer: Email addresss for oauth (@deprecated)
-    def initialize(path_to_key: nil, issuer: nil, path_to_service_account_json: nil)
+    def initialize(path_to_key: nil, issuer: nil, path_to_service_account_json: nil, service_account_json: nil)
```

Comment:
Could this be simplified to only accept the `service_account_json` and do the `File.open` in `make_from_config`?